gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/FirestoreDataStore.java

```diff
@@ -65,6 +66,14 @@
         .subscribeOn(schedulers.io());
   }
 
+  @Cold
+  @Override
+  public  Single<Terms> loadTerms() {
+    return db.terms().getTerm().get()
+        .switchIfEmpty(Single.error(() -> new NotFoundException("No Terms of Service Founds")))
```

Comment:
Consider returning a `Maybe` here and returning an error instead, since terms not being found is an expected/valid state in many cases.

Reply:
NotFound Exception is later used to take the decision whether to redirect directly to the Main Page or not.